src/main/java/com/zaxxer/nuprocess/internal/BasePosixProcess.java

```diff
@@ -616,10 +611,10 @@ public boolean writeStdin(int availability)
          return false;
       }
 
-      if (remainingWrite > 0) {
+      if (inBuffer.hasRemaining()) {
          int wrote;
          do {
-            wrote = LibC.write(fd, inBufferPointer.share(writeOffset), Math.min(remainingWrite, availability));
+            wrote = LibC.write(fd, inBuffer, Math.min(availability, inBuffer.remaining()));
```

Comment:
I do have a question about this change.  How does LibC.write() know what offset to start the write from?  Does JNA automatically calculate the offset based on the fact that ``inBuffer`` is a ByteBuffer?